include/kllvm/ast/AST.h

```diff
@@ -1023,9 +1023,9 @@ class kore_definition {
   [[nodiscard]] std::list<kore_axiom_declaration *> const &get_axioms() const {
     return axioms_;
   }
-  [[nodiscard]] kore_axiom_declaration *
+  [[nodiscard]] sptr<kore_axiom_declaration>
   get_axiom_by_ordinal(size_t ordinal) const {
-    return ordinals_.at(ordinal);
+    return std::make_shared<kore_axiom_declaration>(*ordinals_.at(ordinal));
```

Comment:
I think we can have this return a const ref to the underlying ordinal, and use `return_value_policy::reference` (https://pybind11.readthedocs.io/en/stable/advanced/functions.html)

Reply:
I think so, but using a smart pointer here isn't safer and a better design option?

Reply:
A smart pointer would be safer and a better design option when compared to a _regular pointer_, but a `const&` reference is better than either.

Reply:
Perfect then. Thanks for pointing that out!